pumpkin/src/client/player_packet.rs

```diff
@@ -32,13 +33,26 @@ use pumpkin_protocol::{
     },
 };
 use pumpkin_world::block::{block_registry::get_block_by_item, BlockFace};
+use thiserror::Error;
 
 use super::PlayerConfig;
 
 fn modulus(a: f32, b: f32) -> f32 {
     ((a % b) + b) % b
 }
 
+#[derive(Debug, Error)]
+pub enum PlayerError {
```

Comment:
This is a horrible name. This can be literally everything. I think something like "BlockPlaceError" would be much better suiteted